.github/workflows/tests.yml

```diff
@@ -494,6 +494,211 @@ jobs:
+          chmod +x "$(go env GOPATH)/bin/mc"
+
+      - name: Download system test dependencies
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: system-test-deps
+          merge-multiple: true
+          path: /home/runner/go/bin
+
+      - name: Set exec perms on LXD binaries
+        run: |
+          set -eux
+          ls -lR /home/runner/go/bin/
+          chmod uog+x /home/runner/go/bin/*
+
+      - name: Download docs (for the objects.inv.txt file to be available for the e2e tests)
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: documentation
+          merge-multiple: true
+          path: doc/_build
+
+      - name: Run LXD daemon
+        run: |
+          set -eux
+          echo "root:1000000:1000000000" | sudo tee /etc/subuid /etc/subgid
+          export CGO_CFLAGS="-I/home/runner/work/lxd/lxd-test/vendor/dqlite/include/"
+          export CGO_LDFLAGS="-L/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LD_LIBRARY_PATH="/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LXD_DOCUMENTATION="/home/runner/work/lxd/lxd/doc/_build/"
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo rm -rf /var/lib/lxd
+          sudo -E PATH=${PATH} LD_LIBRARY_PATH=${LD_LIBRARY_PATH} $(go env GOPATH)/bin/lxd --group sudo &
+
+      - name: Setup LXD
+        shell: bash
+        run: |
+          set -eux
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo -E LXD_DIR=/var/lib/lxd lxc storage create default zfs
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default root disk path=/ pool=default
+          sudo -E LXD_DIR=/var/lib/lxd lxc network create local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default eth0 nic network=local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set core.https_address "[::]:8443"
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set cluster.https_address "127.0.0.1"
+          sudo -E LXD_DIR=/var/lib/lxd lxc cluster enable local
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set user.show_permissions=true
+
+      - name: Checkout LXD-UI
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: 'canonical/lxd-ui'
+          ref: main
+          path: lxd-ui
+
+      - name: Install Dotrun
+        run: sudo pip3 install dotrun
+
+      - name: Generate keys for the ui to use
+        run: |
+          . test/includes/certificates.sh
+          export LXD_CONF=lxd-ui/keys
+          mkdir -p $LXD_CONF
+          gen_cert_and_key lxd-ui
+          openssl dhparam -out lxd-ui/keys/dhparams.pem 2048
+          cat lxd-ui/keys/lxd-ui.key lxd-ui/keys/lxd-ui.crt > lxd-ui/keys/lxd-ui.pem
+          cp lxd-ui/keys/lxd-ui.key lxd-ui/keys/lxd-ui.crt.key
```

Comment:
What's that `.crt.key` file for? I find the extension a bit misleading as it contains just the private key, isn't it?

Reply:
I think this is also needed for HAProxy, suspecting a convention for it to be named like that. I might be wrong, but don't have the time now to do an experiment of removing it, then rerunning the workflow and dev server and demo server and see if anything fails.